src/protocol/basics/reshare.rs

```diff
@@ -95,13 +37,28 @@ impl<F: Field> LegacyReshare<F> for MaliciousContext<'_, F> {
 ///    `to_helper`       = (`rand_left`, `rand_right`)     = (r0, r1)
 ///    `to_helper.right` = (`rand_right`, part1 + part2) = (r0, part1 + part2)
 #[async_trait]
-pub trait Reshare<C: Context, B: RecordBinding> {
-    type Output: Sized;
-
+pub trait Reshare<C: Context, B: RecordBinding>: Sized {
     async fn reshare<'fut>(
+        &self,
+        ctx: C,
+        record_binding: B,
+        to_helper: Role,
+    ) -> Result<Self, Error>
+    where
+        C: 'fut;
+}
+
+// Variant of Reshare implemented for types like [T] where resharing doesn't return `Self`.  This is
+// just a more general version of `Reshare`, but a bunch of consumers that don't otherwise care
+// would need to specify an associated type bound `T: Reshare<Output = T>` to use this more general
+// version.
```

Comment:
👍  If only there was a way to provide a default for an associated type... https://github.com/rust-lang/rust/issues/29661